warehouse_ros_mongo/mongo_wrapper_ros.py

```diff
@@ -30,55 +32,61 @@ def print_help_message():
-dbpath = os.path.expanduser(path_param)
-overwrite = rospy.get_param('~overwrite', False)
+dbpath = os.path.expanduser(path_param.value)
+overwrite = node.get_parameter_or('overwrite', False)
 
 if '--repair' in sys.argv:
-    rospy.loginfo("Repairing database")
+    node.get_logger().info("Repairing database")
     lock_file = '{0}/mongod.lock'.format(dbpath)
     if os.path.exists(lock_file):
-        rospy.loginfo("  Removing lock file")
+        node.get_logger().info("  Removing lock file")
         os.remove(lock_file)
     check_call(['mongodb', 'mongod', '--repair', '--dbpath', dbpath])
-    rospy.loginfo("  Successfully repaired.")
+    node.get_logger().info("  Successfully repaired.")
     sys.exit(0)
 
 # The defaults here should match the ones used by each client library
-port = rospy.get_param('warehouse_port', 27017)
-host = rospy.get_param('warehouse_host', 'localhost')
+port = node.get_parameter_or('warehouse_port', Parameter('', Parameter.Type.INTEGER, 27017))
+host = node.get_parameter_or('warehouse_host', Parameter('', Parameter.Type.STRING, 'localhost'))
 
 if overwrite and os.path.exists(dbpath):
-    rospy.loginfo('Removed existing db at %s', dbpath)
+    node.get_logger().info('Removed existing db at {}'.format(dbpath))
     shutil.rmtree(dbpath)
 
 while not os.path.exists(dbpath):
-    rospy.loginfo('{0} does not exist; creating it.'.format(dbpath))
+    node.get_logger().info('{0} does not exist; creating it.'.format(dbpath))
     try:
         os.makedirs(dbpath)
         break
     except OSError as e:
         if dbpath != default_db_path:
-            rospy.logerr('{0}. Trying {1} instead.'.format(str(e), default_db_path))
+            node.get_logger().error('{0}. Trying {1} instead.'.format(str(e), default_db_path))
             dbpath = default_db_path
         else:
-            rospy.logfatal(e)
+            node.get_logger().fatal(e)
             raise e
 
-rospy.loginfo('Starting mongodb with db location {0} listening on {2}:{1}'.\
-              format(dbpath, port, host))
+node.get_logger().info('Starting mongodb with db location {0} listening on {2}:{1}'.\
+              format(dbpath, port.value, host.value))
 
 try:
-    check_call("mongod --dbpath {} --port {}".format(dbpath, port).split())
+    check_call("mongod --dbpath {} --port {}".format(dbpath, port.value).split())
 except CalledProcessError as e:
     if e.returncode==12:
-        rospy.loginfo("Ignoring mongod's non-standard return code of 12")
+        node.get_logger().info("Ignoring mongod's non-standard return code of 12")
     else:
-        rospy.logerr("Mongo process exited with error code {0}".format(e.returncode))
+        node.get_logger().error("Mongo process exited with error code {0}".format(e.returncode))
 except OSError as e:
-    rospy.logerr("Execution failed: %s", e)
+    node.get_logger().error("Execution failed: {}".format(e))
+
+rclpy.spin(node)
+node.destroy_node()
```

Comment:
Just curious, do we really have to call these explicitly?

Reply:
No, this is not necessary. These lines are removed.